src/main/java/org/jabref/gui/importer/actions/AppendDatabaseAction.java

```diff
@@ -159,32 +159,31 @@ public void action() {
             }
             filesToOpen.addAll(chosen);
 
-            // Run the actual open in a thread to prevent the program locking until the file is loaded.
-            JabRefExecutorService.INSTANCE.execute(
-                    () -> openIt(dialog.importEntries(), dialog.importStrings(), dialog.importGroups(), dialog.importSelectorWords()));
-        }
-    }
+            if (filesToOpen.isEmpty()) {
+                return;
+            }
 
-    private void openIt(boolean importEntries, boolean importStrings, boolean importGroups,
-                        boolean importSelectorWords) {
-        if (filesToOpen.isEmpty()) {
-            return;
-        }
-        for (Path file : filesToOpen) {
-            try {
-                Globals.prefs.put(JabRefPreferences.WORKING_DIRECTORY, file.getParent().toString());
-                // Should this be done _after_ we know it was successfully opened?
-                ParserResult parserResult = OpenDatabase.loadDatabase(file.toFile(),
-                        Globals.prefs.getImportFormatPreferences(), Globals.getFileUpdateMonitor());
-                AppendDatabaseAction.mergeFromBibtex(panel, parserResult, importEntries, importStrings, importGroups,
-                        importSelectorWords);
-                panel.output(Localization.lang("Imported from library") + " '" + file + "'");
-            } catch (IOException | KeyCollisionException ex) {
-                LOGGER.warn("Could not open database", ex);
-
-                dialogService.showErrorDialogAndWait(Localization.lang("Open library"), ex);
+            for (Path file : filesToOpen) {
+                // Run the actual open in a thread to prevent the program locking until the file is loaded.
+                BackgroundTask.wrap(() -> openIt(file, dialog.importEntries(), dialog.importStrings(), dialog.importGroups(), dialog.importSelectorWords()))
+                              .onSuccess(fileName -> { dialogService.notify(Localization.lang("Imported from library") + " '" + fileName + "'");})
```

Comment:
singleLineLambdas usually don't need curly braces

Reply:
fixed it :)